src/global_logic.js

```diff
@@ -930,7 +948,145 @@ module.exports.calcHaisuiValue = function (haisuiType, haisuiAmount, haisuiSLv,
+                value = -0.0132 * remainHP + 0.0299;
+            } else {
+                value = -0.0734 * remainHP + 0.06;
+            }
+        } else if (haisuiAmount == "3") {
+            if (remainHP > 0.75 && remainHP <= 1) {
+                value = 0.01;
+            } else if  (remainHP > 0.5 && remainHP <= 0.75) {
+                value = -0.0148 * remainHP + 0.0411;
+            } else {
+                value = -0.0826 * remainHP + 0.075;
+            }
+        } else if (haisuiAmount == "4") {
+            if (remainHP > 0.75 && remainHP <= 1) {
+                value = 0.01;
+            } else if  (remainHP > 0.5 && remainHP <= 0.75) {
+                value = -0.02 * remainHP + 0.045;
+            } else {
+                value = -0.11 * remainHP + 0.09;
+            }
+        } else if (haisuiAmount == "5") {
+            if (remainHP > 0.75 && remainHP <= 1) {
+                value = 0.01;
+            } else if  (remainHP > 0.5 && remainHP <= 0.75) {
+                value = -0.016 * remainHP + 0.052;
+            } else {
+                value = -0.112 * remainHP + 0.1;
+            }
+        } else if (haisuiAmount == "6") {
+            if (remainHP > 0.75 && remainHP <= 1) {
+                value = 0.01;
+            } else if  (remainHP > 0.5 && remainHP <= 0.75) {
+                value = -0.02 * remainHP + 0.055;
+            } else {
+                value = -0.13 * remainHP + 0.11;
+            }
+        } else if (haisuiAmount == "7") {
+            if (remainHP > 0.75 && remainHP <= 1) {
+                value = 0.01;
+            } else if  (remainHP > 0.5 && remainHP <= 0.75) {
+                value = -0.0268 * remainHP + 0.0601;
+            } else {
+                value = -0.1466 * remainHP + 0.12;
+            }
+        } else if (haisuiAmount == "8") {
+            if (remainHP > 0.75 && remainHP <= 1) {
+                value = 0.01;
+            } else if  (remainHP > 0.5 && remainHP <= 0.75) {
+                value = -0.024 * remainHP + 0.068;
+            } else {
+                value = -0.138 * remainHP + 0.125;
+            }
+        } else if (haisuiAmount == "9") {
+            if (remainHP > 0.75 && remainHP <= 1) {
+                value = 0.01;
+            } else if  (remainHP > 0.5 && remainHP <= 0.75) {
+                value = -0.0292 * remainHP + 0.0719;
+            } else {
+                value = -0.1604 * remainHP + 0.1375;
+            }
+        } else if (haisuiAmount == "10") {
+            if (remainHP > 0.75 && remainHP <= 1) {
+                value = 0.01;
+            } else if  (remainHP > 0.5 && remainHP <= 0.75) {
+                value = -0.0332 * remainHP + 0.0749;
+            } else {
+                value = -0.1834 * remainHP + 0.15;
+            } 
+            
+        return value;
+        }
+            
+     if (haisuiType == 'EXLBKonshin' || haisuiType == 'LBKonshin') 
+        if (haisuiAmount == "1") {
```

Comment:
Instead of checking the remainHP border, I suggest set a cap `Math.min(0.03, 0.03*remainHP+0.01);`
then the formula code can be:
```jsx
swtich (haisuiAmount) {
case 1: return Math.min(0.03, 0.03*remainHP+0.01);
case 2: return 0.03*remainHP+0.02;
:
}
```

NOTE: I am not sure here, `case 1:` or `case "1"` JavaScript convert the type number or string implicitly.

Reply:
`case 2: return Math.min(0.04, 0.03*remainHP+0.01);` it make easy to read the code by add cap to all cases. even when hp is 100% just reach the limit.